app/controllers/runtime/organizations_controller.rb

```diff
@@ -326,10 +314,13 @@ def add_role(guid, role, user_id, username)
       org = find_guid_and_validate_access(:update, guid)
 
       if config.get(:perm, :enabled)
-        r = @perm_client.create_role("org-#{role}-#{guid}")
         actor = CloudFoundry::Perm::V1::Models::Actor.new(id: user_id, issuer: SecurityContext.token['iss'])
 
-        @perm_client.assign_role(actor, r.id)
+        begin
+          @perm_client.assign_role(actor, "org-#{role}-#{guid}")
+        rescue GRPC::AlreadyExists
```

Comment:
Could this be done at the client level?

Reply:
Talked about it in person, this is meant to be a generic client.  They will consider making a CC wrapper class to handle these.